docs/DeveloperGuide.md

```diff
@@ -1,38 +1,536 @@
 # Developer Guide
 
+## Table of Contents
+* [Acknowledgements](#acknowledgements)
+* [Design and Implementation](#design--implementation)
+  * [Application Lifecycle](#application-lifecycle)
+  * [Parser Component](#parser-component)
+  * [Command Component](#command-components)
+  * [Storage Component](#storage-component)
+  * [UI Component](#ui-component)
+* [Product Scope](#product-scope)
+* [Non Functional Requirements](#non-functional-requirements)
+* [Glossary](#glossary)
+* [Instructions for Manual Testing](#instructions-for-manual-testing)
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+1. For the main structure of the program, we have adapted the project structure
+from the Individual Project of Man Juncheng at [Link](https://github.com/spinoandraptos/ip/tree/master) </li>
+2. For JUnit testing, we have adapted the testing codes from the AddressBook level-2
+codes at [Link](https://github.com/se-edu/addressbook-level2)</li>
+3. For the developer's guide, we have referenced the developer's guide from AddressBook
+level-3 [Link](https://se-education.org/addressbook-level3/DeveloperGuide.html)
+
+# Design & implementation
+
+## Overall Architecture
+The main execution of the QuizHub application will concern 4 components which
+are the QuizHub, Ui, Parser and Command packages.
+
+![](./UML/architecture.jpg)
+
+`Quizhub` is the "main" class, which is responsible for starting the program,
+initialising all other objects, executing commands, and shutting down.
+
+`UI` is the component that interacts with the user, taking inputs and displaying
+the results of the executed commands.
+
+`Parser` acts as a multiplexer to determine which command to run, and what parameters
+it has, based on the user input.
+
+`Commands` refer to a package of individual commands with complex and specific
+logic, which is later executed in Quizhub and displayed by `UI`.
+
+`Storage` is the class through which questions can be stored on the hard drive and `Utility Classes`
+refer to any miscellaneous utility classes used by all the components.
+
+## Application Lifecycle
+
+![](./UML/lifecycle.jpg)
```

Comment:
Is the specific command instance still relevant after the execution? You might consider marking the end of its lifetime with a red X at the bottom of the activation bar.
<img width="213" alt="Screenshot 2023-11-02 at 17 19 57" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/62311026/3fb3c53c-b8d8-4a3b-962b-5c2e1aa28cb2">